packages/node_modules/@webex/webex-core/src/lib/services/service-catalogs.js

```diff
@@ -41,6 +54,54 @@ export default class ServiceCatalogs {
     this.hosts = [];
   }
 
+  /**
+   * An active, local, and priority mapped record of the current
+   * {@link ServiceCatalog#hosts}.
+   *
+   * @public
+   * @memberof ServiceCatalog
+   * @type {Record<string, string>}
+   */
+  get map() {
+    // Get a list of active, local, and priority-mapped hosts.
+    return this.find({active: true, local: true, priority: true}).reduce(
+      (map, host) => {
+        // Generate a new object to assign the existing map.
+        const hostReference = {};
+
+        // Assign the key:value pair for the service and url.
+        hostReference[host.service] = host.url;
+
+        // Assign the reference to the map.
+        Object.assign(map, hostReference);
```

Comment:
Lets do a immutable+rest instead of assign a param
```js
return {...map, ...hostReference}
```

Reply:
Curious which is better for performance?

Reply:
https://jsperf.com/object-assign-vs-spreading

Reply:
Implemented these changes.